File: src/main/java/com/example/Jobportal/model/Apply.java
package com.example.Jobportal.model;


import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.*;

import javax.persistence.*;

@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@ToString
@Table(name = "apply")
public class Apply {

    @EmbeddedId
    private CompositeKey id;

    @ManyToOne
    @JsonManagedReference
    @MapsId("candidateId")
    @JoinColumn(name = "candidate_id")
    private Candidate candidateApply;

    @ManyToOne
    @JsonManagedReference
    @MapsId("jobId")
    @JoinColumn(name = "job_id")
    private Job jobApply;

    private String status;
}

File: app/src/main/java/com/example/joblist/entities/apply/Apply.kt
package com.example.joblist.entities.apply

import com.example.joblist.entities.Candidate
import com.example.joblist.entities.Job

data class Apply(
    val applyDate: String,
    val candidateApply: Candidate,
    val id: ApplyId,
    val jobApply: Job,
    val status: String
)
